modules/apps/transfer/types/v3/packet_test.go

```diff
@@ -0,0 +1,571 @@
+		},
+		{
+			"failure: invalid zero amount",
+			NewFungibleTokenPacketData(
+				[]*Token{
+					{
+						Denom:  "atom/pool",
+						Amount: "0",
+						Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+					},
+				},
+				sender,
+				receiver,
+				"",
+			),
+			types.ErrInvalidAmount,
+		},
+		{
+			"failure: invalid negative amount",
+			NewFungibleTokenPacketData(
+				[]*Token{
+					{
+						Denom:  "atom/pool",
+						Amount: "-100",
+						Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+					},
+				},
+				sender,
+				receiver,
+				"",
+			),
+			types.ErrInvalidAmount,
+		},
+		{
+			"failure: invalid large amount",
+			NewFungibleTokenPacketData(
+				[]*Token{
+					{
+						Denom:  "atom/pool",
+						Amount: invalidLargeAmount,
+						Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+					},
+				},
+				sender,
+				receiver,
+				"memo",
+			),
+			types.ErrInvalidAmount,
+		},
+		{
+			"failure: missing sender address",
+			NewFungibleTokenPacketData(
+				[]*Token{
+					{
+						Denom:  "atom/pool",
+						Amount: "1000",
+						Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+					},
+				},
+				"",
+				receiver,
+				"memo",
+			),
+			ibcerrors.ErrInvalidAddress,
+		},
+		{
+			"failure: missing recipient address",
+			NewFungibleTokenPacketData(
+				[]*Token{
+					{
+						Denom:  "atom/pool",
+						Amount: "1000",
+						Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+					},
+				},
+				sender,
+				"",
+				"",
+			),
+			ibcerrors.ErrInvalidAddress,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := tc.packetData.ValidateBasic()
+
+		expPass := tc.expErr == nil
+		if expPass {
+			require.NoError(t, err, tc.name)
+		} else {
+			require.ErrorContains(t, err, tc.expErr.Error(), tc.name)
+		}
+	}
+}
+
+func TestGetPacketSender(t *testing.T) {
+	testCases := []struct {
+		name       string
+		packetData FungibleTokenPacketData
+		expSender  string
+	}{
+		{
+			"non-empty sender field",
+			NewFungibleTokenPacketData(
+				[]*Token{
+					{
+						Denom:  "atom/pool",
+						Amount: "1000",
+						Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+					},
+				},
+				sender,
+				receiver,
+				"",
+			),
+			sender,
+		},
+		{
+			"empty sender field",
+			NewFungibleTokenPacketData(
+				[]*Token{
+					{
+						Denom:  "atom/pool",
+						Amount: "1000",
+						Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+					},
+				},
+				"",
+				receiver,
+				"abc",
+			),
+			"",
+		},
+	}
+
+	for _, tc := range testCases {
+		require.Equal(t, tc.expSender, tc.packetData.GetPacketSender(types.PortID))
+	}
+
+	packetData := NewFungibleTokenPacketData(
+		[]*Token{
+			{
+				Denom:  "atom/pool",
+				Amount: "1000",
+				Trace:  []string{"transfer/channel-0", "transfer/channel-1"},
+			},
+		},
+		sender,
+		receiver,
+		"",
+	)
+	require.Equal(t, sender, packetData.GetPacketSender(types.PortID))
```

Comment:
This code can be removed after the changes in https://github.com/cosmos/ibc-go/pull/6126/commits/1f6c780a67e2a0b8b427289d16b3e0960c86ce34, right?